apps/accounts/models.py

```diff
@@ -272,18 +274,19 @@ def save(self, **kwargs):
 
         super(ActivationKey, self).save(**kwargs)
 
-        # send an email with activation url
-        activation_link = '%s%s' % (get_hostname(),
-                                    reverse('activation_verify',
-                                            args=(self.key,)))
-        mailer = Mailer(subject='Verify Your Blue Button 2.0 Developer Sandbox Account',
-                        template_text='email/email-activate.txt',
-                        template_html='email/email-activate.html',
-                        to=[self.user.email, ],
-                        context={"ACTIVATION_LINK": activation_link})
-        mailer.send()
-        logger.info("Activation link sent to {} ({})".format(self.user.username,
-                                                             self.user.email))
+        if switch_is_active('outreach_email'):
```

Comment:
it would be ideal if these could be decorators.

e.g.
```python
outreach_welcome_email(<args>)
...

@waffle_switch('outreach_email')
def outreach_welcome_email(<args>):
    ...
```

It would require making a decorator that looks something like:
```python
def waffle_switch(switch_name):
    def decorator(func):
        @wraps(func, assigned=available_attrs(func))
        def _wrapped_func(*args, **kwargs):
            if switch_name.startswith('!'):
                active = not switch_is_active(switch_name[1:])
            else:
                active = switch_is_active(switch_name)

            if not active:
                return
            return func(*args, **kwargs)
        return _wrapped_func
    return decorator
```

Reply:
This is a great idea! I added a decorator called waffle_function_switch. The code you provided worked for this.